tonic/models/vic/grid_params.py

```diff
@@ -917,6 +1150,34 @@ def grid_params(soil_dict, target_grid, snow_dict, veglib_dict, veg_dict,
                 for y, x in pyzip(ymask, xmask):
                     out_dict[var][:, :, y, x] = fill_val
 
+            elif mydict[var].ndim == 4:
+                j = mydict[var].shape[1]
+                k = mydict[var].shape[2]
+                l = mydict[var].shape[3]
+                out_dict[var] = np.ma.zeros((j, k, l, ysize, xsize),
+                                            dtype=dtype)
+                for jj in pyrange(j):
+                    for kk in pyrange(k):
+                        for ll in pyrange(l):
+                            out_dict[var][jj, kk, ll, yi, xi] = mydict[var][:, jj, kk, ll]
+                for y, x in pyzip(ymask, xmask):
+                    out_dict[var][:, :, :, y, x] = fill_val
```

Comment:
I don't think this needs to be done in a for loop.  I think you can just do

```Python
out_dict[var][:, :, :, ymask, xmask] = fill_val
```

better yet, why not just prefill `out_dict[var]` (see line 1157):

```Python
out_dict[var] = np.full((j, k, l, ysize, xsize), fill_val, dtype=dtype)
```